packages/sign_in_with_apple/sign_in_with_apple_platform_interface/pubspec.yaml

```diff
@@ -0,0 +1,22 @@
+name: sign_in_with_apple_platform_interface
+description: A common platform interface for the sign_in_with_apple plugin.
+repository: https://github.com/aboutyou/dart_packages/tree/master/packages/sign_in_with_apple/sign_in_with_apple_platform_interface
+version: 1.0.0
```

Comment:
Is there any guideline regarding versioning these platform interface/implementation packages?

E.g., keep all packages on the same version (maybe also just the major version would be enough)?

Reply:
I think the overall guideline is just to be 100% compliant with semver and not have any breaking changes without a major version upgrade.

Because in practice it's expected that the interface is implementation by third party packages as well, and no all (e.g. platform, native mobile, web) would be updated at the same time. So it's important that plugins from different sources are all compatible with the same major version of the interface.

(That's also the reason why there is the `extends` pattern, in which case platform plugins automatically "get the new methods", though in practice they would just throw until it's implemented. But that's by design / the Flutter guidelines.)